### PR TITLE
Prefaces tab

### PR DESCRIPTION
**Pull request by miri**
#66

### CODE REVIEW

**Comment by miri** [comment]
image

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 1]
```diff
@@ -107,6 +117,27 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
++ */
+function haskala_book_prefaces_tab_additional_info($book_id, $person_id) {
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', array('production'))
+    ->fieldCondition('field_producer', 'target_id', $person_id)
+    ->fieldCondition('field_book', 'target_id', $book_id);
+  $result = $query->execute();
+  if (!empty($result)) {
+    $production_id = array_keys($result['node'])[0];
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Change to ```php $production_id = reset($result['node']);```

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 2]
```diff
@@ -107,6 +117,27 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
++ * Prefaces tab - role in book production of preface writer
++ */
+function haskala_book_prefaces_tab_additional_info($book_id, $person_id) {
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', array('production'))
+    ->fieldCondition('field_producer', 'target_id', $person_id)
+    ->fieldCondition('field_book', 'target_id', $book_id);
+  $result = $query->execute();
+  if (!empty($result)) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Change to: 
```
if(empty($result['node']) { 
  return;
}
```

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 3]
```diff
@@ -107,6 +117,27 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
 }
 
 /**
++ * Prefaces tab - role in book production of preface writer
++ */
+function haskala_book_prefaces_tab_additional_info($book_id, $person_id) {
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', array('production'))
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: array('production') can be 'production'

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 4]
```diff
@@ -107,6 +117,27 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
+    ->fieldCondition('field_producer', 'target_id', $person_id)
+    ->fieldCondition('field_book', 'target_id', $book_id);
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Also check the node status is NODE_PUBLISHED

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 5]
```diff
@@ -69,6 +69,16 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
       if ($wrapper->$field_name->value()) {
         $field = render($node_view[$field_name]);
         $tabs_fields[$tab->label][] = $field;
+
+        // Role of preface writer
+        if ($field_name == "field_writer_of_preface") {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Use single quotes

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 6]
```diff
@@ -69,6 +69,16 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
+          $production_id = haskala_book_prefaces_tab_additional_info($node->nid, $wrapper->$field_name->value()->nid);
+          if (isset($production_id)) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: $production_id is always set at this point (Also if it's NULL). You probably want to check ``` if ($production_id) {```

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 7]
```diff
@@ -107,6 +117,28 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'production')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('field_producer', 'target_id', $person_id)
+    ->fieldCondition('field_book', 'target_id', $book_id);
+  $result = $query->execute();
+  if (empty($result['node'])) {
+    return;
+  }
+  else {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: No need for the else. Change to 
``` return reset($result['node'])->nid; ```

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 8]
```diff
@@ -107,6 +117,25 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
 }
 
 /**
++ * Prefaces tab - role in book production of preface writer
++ */
+function haskala_book_prefaces_tab_additional_info($book_id, $person_id) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: - Change the function name according to what it actually does - Perhaps haskala_book_get_production()
- Change $book_id to $book_nid, same for other vairables

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 9]
```diff
@@ -107,6 +117,25 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
 }
 
 /**
++ * Prefaces tab - role in book production of preface writer
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Add @params and @return

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc [hunk 10]
```diff
@@ -107,6 +117,25 @@ function haskala_book_book_fields_content_type_render($subtype, $conf, $args, $c
...
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'production')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('field_producer', 'target_id', $person_id)
+    ->fieldCondition('field_book', 'target_id', $book_id);
+  $result = $query->execute();
+  if (empty($result['node'])) {
+    return;
+  }
+  return $production_id = reset($result['node'])->nid;
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/book_fields.inc]: Change to: ```return reset($result['node'])->nid;```

**Comment by bitamar** [comment]
Merged

**Pull request closed without merging**